dask/dataframe/core.py

```diff
@@ -360,19 +360,58 @@ def map_partitions(self, func, *args, **kwargs):
 
         Examples
         --------
+        Given a DataFrame, Series, or Index, such as:
 
-        When tuple is passed as meta, the result will be Series.
+        >>> import dask.dataframe as dd
+        >>> df = pd.DataFrame({'x': [1, 2, 3, 4, 5],
+        ...                    'y': [1., 2., 3., 4., 5.]})
+        >>> ddf = dd.from_pandas(df, npartitions=2)
 
-        >>> df.map_partitions(lambda df: df.x + 1, meta=('x', 'i8'))  # doctest: +SKIP
+        One can use ``map_partitions`` to apply a function on each partition.
+        Extra arguments and keywords can optionally be provided, and will be
+        passed to the function after the partition.
 
-        When dict is passed as meta, the result will be DataFrame.
+        Here we apply a function with arguments and keywords to a DataFrame,
+        resulting in a Series:
 
-        >>> df.map_partitions(lambda df: df.head(), meta={'x': 'i8', 'y': 'i8'})  # doctest: +SKIP
+        >>> def myadd(df, a, b=1):
+        ...     return df.x + df.y + a + b
+        >>> res = ddf.map_partitions(myadd, 1, b=2)
+        >>> res.dtype
+        dtype('float64')
 
-        Note that you can also pass in the DataFrame/Series itself, if the
-        metadata doesn't change:
+        By default, dask tries to infer the output metadata by running your
+        provided function on some fake data. This works well in many cases, but
+        can sometimes be expensive, or even fail. To avoid this, you can
+        manually specify the output metadata with the ``meta`` keyword. This
+        can be specified in many forms, for more information see
+        ``dask.dataframe.utils.make_meta``.
 
-        >>> df.map_partitions(lambda df: df.head(), meta=df)  # doctest: +SKIP
+        Here we specify the output is a Series with no name, and dtype
+        ``float64``:
+
+        >>> res = ddf.map_partitions(myadd, 1, b=2, meta=(None, 'f8'))
```

Comment:
Is the description of `meta` in parameters, and the above prose:

> This can be specified in many forms, for more information see ``dask.dataframe.utils.make_meta``

not good enough? I wanted to illustrate what I envision to be the most common use inputs.